src/main/resources/workerScripts/watchdog.lua

```diff
@@ -102,18 +105,90 @@ local function inspectLightKeeper()
     end
 end
 
+local function runRecoveryProcess()
+    debugMessage("Start redis restart recovery process on server "..serverName)
+    redis.call('SET', redisRecoveryKey,inProgressPrefix..serverName)
+    return "true"
+end
+
+local function checkIfRecoveryRequired()
+    local runRecovery = "false"
 
-------------------------------------------------
+    local redisRecoveryValue = redis.call('GET', redisRecoveryKey)
 
+    if (not redisRecoveryValue) then -- redis restart was detected
+        -- start recovery process
+        runRecovery = runRecoveryProcess();
+    elseif(redisRecoveryValue ~= "READY") then-- redis recovery in progress
+        -- check server running recovery is still alive
+        local recoveryServer = redisRecoveryValue:sub(inProgressPrefix:len()+1)
+        local recoverServerIsAliveKey = isAlivePrefix..recoveryServer;
+        local isAliveLastRun = redis.call('GET', recoverServerIsAliveKey)
+
+        local millisDiff = tonumber(currentTime)- tonumber(isAliveLastRun)
+
+        -- if recovery server is not alive , rerun on current server
+        if millisDiff > lightKeeperPeriod then
+            debugMessage("Recovery server was not active for "..tostring(millisDiff).." millis,rerunning")
+            runRecovery = runRecoveryProcess();
+        end
+    end
+
+    return runRecovery
+end
+
+local function updateRecoveryStatus(recoveryProcessStatus)
+    if(recoveryProcessStatus == "FINISHED") then
+        -- mark recovery as finished
+        debugMessage("Redis restart recovery process has finished on server "..serverName)
+        redis.call('SET', redisRecoveryKey,"READY")
+    elseif (recoveryProcessStatus == "FAILED") then
+        -- allow to other server to run recovery
+        debugMessage("Redis restart recovery process has failed on server "..serverName)
+        redis.call('DEL', redisRecoveryKey)
+    end
+end
 
+local function fixInterruptedRecovery()
+    local redisRecoveryValue = redis.call('GET', redisRecoveryKey)
+
+    if(redisRecoveryValue and redisRecoveryValue ~= "READY") then   -- redis recovery was detected
+        local recoveryServer = redisRecoveryValue:sub(inProgressPrefix:len()+1)
+
+        -- if recovery was running on current server prior to restart , clean out recovery key
+        if(recoveryServer == serverName) then
+            debugMessage("Recovery process was interrupted on "..serverName)
+            redis.call('DEL', redisRecoveryKey)
+        end
+    end
+end
+
+------------------------------------------------
 
 
 if jobName=='watchdog' then
+    timeToRequeueJobsOnInactiveServer = getMillis(ARGV[5])
+    lightKeeperPeriod = getMillis(ARGV[6])
+
     inspectLightKeeper()
 elseif jobName=='requeueJobs' then
     requeueJobs(serverName)
 elseif jobName=='isAlive' then
```

Comment:
We should probably use a verb here :)

Reply:
I've copied it from Thread.isAlive() method. I don't mind changing the name, but can't find better name

Reply:
What about reportAlive?

Reply:
Ok :-)